server/workflow/workflow_server.go

```diff
@@ -238,7 +248,13 @@ func (s *workflowServer) ResubmitWorkflow(ctx context.Context, req *workflowpkg.
 
 func (s *workflowServer) ResumeWorkflow(ctx context.Context, req *workflowpkg.WorkflowResumeRequest) (*v1alpha1.Workflow, error) {
 	wfClient := auth.GetWfClient(ctx)
-	err := util.ResumeWorkflow(wfClient.ArgoprojV1alpha1().Workflows(req.Namespace), req.Name)
+
+	_, err := s.getWorkflow(ctx, req.Namespace, req.Name, metav1.GetOptions{})
```

Comment:
what's this for? check permission?

Reply:
if so - extract to func and reuse?

Reply:
This is already a extracted function, it get the wf and validate the instanceID. I need it to have 2 return values (workflow, error), some invokers need both (e.g. `func GetWorkflow()`), some others only need (error), like this.

I think the name might not be good, I renamed to `getAndValidateWorkflow()`, let me know if you have a better idea to extract it.